### PR TITLE
Cover more bird template branches

### PR DESCRIPTION
**Pull request by heschlie**
This PR gives us coverage of the following bird templates:

- `kdd/mesh/ipip-always`
- `kdd/mesh/ipip-disabled`
- `kdd/mesh/ipip-cross-subnet`
- `etcd/mesh/ipip-always`
- `etcd/mesh/ipip-disabled`
- `etcd/mesh/ipip-cross-subnet`
- `etcd/explicit_peering/global`
- `etcd/explicit_peering/specific_node`

The tests directory mock data is setup to reflect those cases.  The make target `make test` will test all of the above cases by spinning up an etcd server and populating data, and spinning up a k8s API server and populating it with data.  Between each test it will clean up the previous tests data so they do not conflict with each other.

I've also copied the templates from `calico_node` here, if that is not really what we want I can revert back to pulling them in during the test.

### CODE REVIEW

**Comment by tomdee** [comment]
I don't think the templates can me moved as part of this PR, so can you revert that bit.

File: tests/compiled_templates/explicit_peering/global/bird.cfg [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  add paths on;
+  graceful restart;  # See comment in kernel section about graceful restart.
+}
+
+
+# ------------- Global peers -------------
+
+
+
+# For peer /global/peer_v4/10.192.0.3
```
Comment (robbrockbank) [inline tests/compiled_templates/explicit_peering/global/bird.cfg]: I think it's best to test with >1 peer for things like this (if we can)

Comment (heschlie) [inline tests/compiled_templates/explicit_peering/global/bird.cfg]: Should be perfectly doable

**Comment by robbrockbank** [comment]
For the BGP peer tests we should move the default global constants away from their default value:
log level
AS Number

**Comment by robbrockbank** [comment]
Other than the two minor comments above, the current testing looks ok.

**Comment by heschlie** [comment]
@robbrockbank the configs are ready for another look!

File: Makefile [hunk 2]
```diff
@@ -121,3 +121,4 @@ bin/etcdctl:
 .PHONY: clean
 clean:
 	rm -rf bin/*
+	rm -f templates/config/*.cfg
```
Comment (tomdee) [inline Makefile]: What is this line for?

Comment (heschlie) [inline Makefile]: This was to clean the generated files and should have been removed when I backed out the templates dir.

**Comment by heschlie** [comment]
@robbrockbank suggested logging confd output to file, and so it is done.  There is an issue with those logs being written out as root, as we run confd as root, it might be worth trying to set some permissions to the `/etc/calico/` directory so we can run confd as a non-root user.

**Comment by robbrockbank** [comment]
I'm not too fussed about the root logs just at the moment - we can address later if we find it is a problem.

**Comment by heschlie** [comment]
@robbrockbank squashed and ready for merge!

**Pull request merged**